Compile Error in `bubbaloop-node-sdk`: `ReplyBuilder` lacks `res()` method due to Zenoh 1.0.0 API changes

### Background: What I'm Building
I am preparing a proposal for **GSoC 2026** to build a custom computer vision node natively on Bubbaloop using `kornia-rs`. 

While trying to scaffold and bootstrap my new Rust node using the official Bubbaloop CLI and the latest `bubbaloop-node-sdk`, my project immediately failed to compile because of an upstream issue in this repository.

### Describe the Bug
The `bubbaloop-node-sdk` crate fails to compile on the [main](cci:1://file:///Users/namangupta/Documents/GitHub/EdgeLane/bubbaloop-lane-detection/build.rs:0:0-23:1) branch when resolving the latest Zenoh dependencies. The Rust compiler throws `error[E0599]: no method named res found for struct ReplyBuilder` inside the [declare_schema_queryable](cci:1://file:///Users/namangupta/Documents/GitHub/EdgeLane/bubbaloop/crates/bubbaloop-node-sdk/src/schema.rs:2:0-35:1) function.

This occurs because Zenoh 1.0.0 refactored `query.reply()` to return a `Future` instead of a synchronous builder, meaning it must be explicitly `.await`ed, and the older `.res()` method has been removed entirely. 

### To Reproduce
1. Generate a new Bubbaloop node using the Rust template (or clone the repository natively).
2. Run `cargo build -p bubbaloop-node-sdk`
3. Observe the compilation error in `src/schema.rs:23`.

### Expected Behavior
The Node SDK should compile smoothly out-of-the-box on the latest version of Zenoh so downstream custom nodes can be built.

### Proposed Solution
Modify [declare_schema_queryable](cci:1://file:///Users/namangupta/Documents/GitHub/EdgeLane/bubbaloop/crates/bubbaloop-node-sdk/src/schema.rs:2:0-35:1) to handle `query.reply()` asynchronously. Because `zenoh::declare_queryable().callback` takes a synchronous closure, we should wrap the asynchronous reply in a `tokio::spawn(async move { ... })` and clone the [descriptor](cci:1://file:///Users/namangupta/Documents/GitHub/EdgeLane/bubbaloop-lane-detection/src/main.rs:33:4-35:5) before the move to satisfy the `Fn` boundary requirements.

I already have a patch ready for this that allowed my local node block to successfully compile. I will open a Pull Request shortly linking this issue!

--- a/crates/bubbaloop-node-sdk/src/schema.rs
+++ b/crates/bubbaloop-node-sdk/src/schema.rs
@@ -20,9 +20,12 @@ pub async fn declare_schema_queryable(
             let descriptor = descriptor.to_vec();
             move |query| {
                 log::debug!("Schema query received");
-                if let Err(e) = query.reply(&query.key_expr().clone(), descriptor.as_slice()) {
-                    log::warn!("Failed to reply to schema query: {}", e);
-                }
+                let descriptor_clone = descriptor.clone();
+                tokio::spawn(async move {
+                    if let Err(e) = query.reply(&query.key_expr().clone(), descriptor_clone.as_slice()).await {
+                        log::warn!("Failed to reply to schema query: {}", e);
+                    }
+                });
             }
         })
         .await

--- a/docs/node-development.md
+++ b/docs/node-development.md
@@ -351,7 +351,10 @@ async fn main() -> Result<()> {
         .callback({
             let descriptor = DESCRIPTOR.to_vec();
             move |query| {
-                let _ = query.reply(&query.key_expr().clone(), descriptor.as_slice());
+                let descriptor_clone = descriptor.clone();
+                tokio::spawn(async move {
+                    let _ = query.reply(&query.key_expr().clone(), descriptor_clone.as_slice()).await;
+                });
             }
         })
         .await?;

